packages/astro/client-base.d.ts

```diff
@@ -1,4 +1,4 @@
-/// <reference types="vite/types/importMeta" />
+/// <reference path="./import-meta.d.ts" />
```

Comment:
A question: Is there a reason we're not using `vite/client`?

Reply:
There's a few reasons:
- `vite/client` references the dom lib, and in the future we're hopping to remove references to it to improve our completions in the frontmatter
- We need different definitions for our image integration, so our "client types" are splitted off in multiple files to make this easier (when installing the image integration, we ask people to switch the type to something else)
- Historically and still a bit now, we tend to try to hide Vite as much as possible as it confuses users

Reply:
Ah that makes sense. I think it's fair then that we define most of it ourselves. I suppose in the future Vite can split out into more type files too so we can reference most of the `declare module` parts. Thanks for the explanation!